### ISSUE TITLE
onMomentumScrollEnd and OnIndexChanged not firing

### ISSUE THREAD

**Issue by yashaRO** [leecade/react-native-swiper#596]
On Windows 10
- react-native-swiper v1.5.12
- react-native v0.48.4

Oddly enough I have the same code on two pages, with just basic views, and on one it fires, and on the other it doesn't. I'm getting no logs for anything.

I can't fork the code because it's private, but this is the gist of the code.
Working code
```
render() {
		return (
			<View style={styles.container}>
				<Header title={this.state.title} onPress={()=>this.props.navigation.goBack()}/>
				<View style={styles.body}>
					<View style={styles.routes}>
						<Swiper loadMinimal={true} loadMinimalSize={4} showsPagination={false}>
							{this.display()}
						</Swiper>
					</View>
					<View style={styles.alpha}>
						{this.displayAlpha()}
					</View>
				</View>
			</View>
		)
```

Not working code
```
onMomentumScrollEnd = (e, state, context) => {
		Reactotron.log(this.state.List[state.index].title)
		Reactotron.log(true, true)
//		this.setState({
//			title: this.state.subjectList[state.index].title
//		})
	}
	render() {
		return (
			<View style={styles.container}>
				<Header title={this.state.title} onPress={()=>this.props.navigation.goBack()} navigate={this.props.navigation.navigate}/>
				<View style={styles.body}>
					<View style={styles.routes}>
						<Swiper loadMinimal={true} loadMinimalSize={4} showsPagination={false} onMomentumScrollEnd={this.onMomentumScrollEnd}>
							{this.display()}
						</Swiper>
					</View>
					<View style={styles.alpha}>
						{this.displayAlpha()}
					</View>
				</View>
			</View>
		)
	}
```

**Comment by arribbar** [comment]
Do you think you can reproduce the bug with one of our examples ? To do so, you need to fork this project and change one example.
It would help me a lot as I am really busy now...

**Comment by yashaRO** [comment]
I'll try. I know at least onTouchScreenCapture works.

**Comment by yashaRO** [comment]
One thing I just found out, was that the swiper itself isn't working normally, it's loading at index 0 every time.

**Comment by ravirupareliya** [comment]
Yes even in my case also it is not working. It is working fine if we are using static data, but not working when list is dynamic

**Comment by yashaRO** [comment]
Well that was the root of my issue, retrieving the data on the screen caused it not to work, so I retrieved the data earlier and then propped it to the screen and it works. My app runs a bit slower now (at least on emu)

**Comment by changey** [comment]
should this issue be closed then?

**Comment by yashaRO** [comment]
I suppose I can close it, but I wonder if I should make a PR about using static data (preloaded) and/or leave it open  as suggestion for future updates to address

**Comment by changey** [comment]
that would be awesome. thank you! I was looking at the issue description and found that you actually found a method to resolve it

**Comment by xianglinhe** [comment]
@yashaRO you can check the data to see if it's ready before rendering to avoid this issue.
`render() {

        if (this.siteList.length <= 0) {
            return (
                <View style={styles.bgViewStyle}>
                </View>
            )
        }


        return (
            <View style={styles.bgViewStyle}>

                <Swiper paginationStyle={styles.paginationStyle}
                        removeClippedSubviews={false}
                        onIndexChanged={this.onIndexChanged.bind(this)}>

                    {
                        this.siteList.map((item, index) =>
                            <View style={styles.wrapperStyle} key={index}>
                                <Text style={styles.siteNameTextStyle}>{item.name}</Text>
                            </View>
                        )
                    }
                </Swiper>

            </View>
        )
    }`

**Comment by yashaRO** [comment]
Oddly enough, I structured my code like that on some pages, but didn't think to do that on the swiper pages. I'm fine with preloading, but I'll try that method as well.
